src/path_cache.rs

```diff
@@ -938,12 +945,248 @@ impl<N: Neighborhood + Sync> PathCache<N> {
+
+            let chunk_index = self.get_chunk_index(cp);
+            let chunk = get_chunk_mut!(self, chunk_index);
+
+            for id in removed {
+                chunk.nodes.remove(&id);
+                self.nodes.remove_node(id); // This is where the majority of time is spent here
+            }
+        }
+
+        println!(
+            "time to remove nodes of sides in renew: {:?}",
+            Instant::now() - timer
+        );
+
+        // remove all Paths in changed chunks
+        for cp in dirty.keys() {
+            let chunk_index = self.get_chunk_index(*cp);
+            let chunk = get_chunk!(self, chunk_index);
+            for id in chunk.nodes.iter() {
+                self.nodes[*id].edges.clear();
+            }
+        }
+
+        let timer = Instant::now();
+        // recreate sides in renew
+        for (&cp, sides) in renew.iter() {
+            let mut candidates = PointSet::default();
+            let chunk_index = self.get_chunk_index(cp);
+            let chunk = get_chunk!(self, chunk_index);
+
+            for dir in Dir::all() {
+                if sides[dir.num()] != Renew::No {
+                    chunk.calculate_side_nodes(
+                        dir,
+                        (self.width, self.height),
+                        &get_cost,
+                        self.config,
+                        &mut candidates,
+                    );
+                }
+            }
+
+            // Only include nodes that aren't already part of the map
+            let candidates: PointSet = candidates
+                .iter()
+                .filter(|&&pos| self.nodes.id_at(pos).is_none())
+                .map(|&a| a)
+                .collect();
+
+            if candidates.is_empty() {
+                continue;
+            }
+
+            let all_nodes = &mut self.nodes;
+            let nodes = candidates
+                .into_iter()
+                .map(|p| all_nodes.add_node(p, get_cost(p) as usize))
+                .to_vec();
+
+            if !dirty.contains_key(&cp) {
+                let chunk = get_chunk_mut!(self, chunk_index);
+                chunk.add_nodes(
+                    nodes,
+                    &get_cost,
+                    &self.neighborhood,
+                    &mut self.nodes,
+                    &self.config,
+                );
+            } else {
+                for id in nodes {
+                    let chunk = get_chunk_mut!(self, chunk_index);
+                    chunk.nodes.insert(id);
+                }
+            }
+        }
+
+        println!("time to recreates sides: {:?}", Instant::now() - timer);
+
+        let timer = Instant::now();
+
+        let dirty_indices: hashbrown::HashSet<usize> = dirty
+            .keys()
+            .map(|(x, y)| self.get_chunk_index((*x, *y)))
+            .collect();
+
+        let mid_timer = Instant::now();
+
+        let paths: Vec<HashMap<u32, HashMap<(usize, usize), Path<(usize, usize)>>>> = {
```

Comment:
```suggestion
        let paths: Vec<_> = {
```
or
`Vec<NodeIDMap<PointMap<Path<Point>>>>`